base/bool.jl

```diff
@@ -3,6 +3,7 @@
 ## boolean conversions ##
 
 convert(::Type{Bool}, x::Bool) = x
+convert(::Type{Bool}, x::Float16) = x==0 ? false : x==1 ? true : throw(InexactError())
```

Comment:
Actually, it was not! There was an ambiguity error from convert after removing this, so it needs to stay, as it is more specific than the other two ( can't remember where they were defined, but you can rebuild without this line and check).